zebra-state/src/service/finalized_state/zebra_db/shielded.rs

```diff
@@ -243,14 +276,6 @@ impl DiskWriteBatch {
         self.zs_insert(sapling_anchors, sapling_root, ());
         self.zs_insert(orchard_anchors, orchard_root, ());
 
-        // Update the trees in state
-        let current_tip_height = *height - 1;
-        if let Some(h) = current_tip_height {
-            self.zs_delete(sprout_note_commitment_tree_cf, h);
-            self.zs_delete(sapling_note_commitment_tree_cf, h);
-            self.zs_delete(orchard_note_commitment_tree_cf, h);
-        }
-
         self.zs_insert(
```

Comment:
This is a follow-up from the engineering sync today. This change is out of scope, but I wanted to add a TODO.

```suggestion
        // TODO: if we ever need concurrent read-only access to the sprout tree,
        // store it by `()`, not height. Otherwise, we could access a deleted height.
        self.zs_insert(
```